sound/soc/soc-dapm.c

```diff
@@ -4464,9 +4464,30 @@ void snd_soc_dapm_connect_dai_link_widgets(struct snd_soc_card *card)
 			for_each_rtd_codec_dais(rtd, i, codec_dai)
 				dapm_connect_dai_pair(card, rtd, codec_dai,
 						      asoc_rtd_to_cpu(rtd, i));
+		} else if (rtd->dai_link->num_codecs > rtd->dai_link->num_cpus &&
+			   rtd->dai_link->codec_ch_maps) {
+			int cpu_id;
+
+			for_each_rtd_codec_dais(rtd, i, codec_dai) {
+				cpu_id = rtd->dai_link->codec_ch_maps[i].connected_cpu_id;
+				if (cpu_id >= rtd->dai_link->num_cpus) {
+					dev_err(card->dev,
+						"dai_link %s cpu_id %d too large, num_cpus is %d\n",
+						rtd->dai_link->name, cpu_id,
+						rtd->dai_link->num_cpus);
+					continue;
+				}
+				dapm_connect_dai_pair(card, rtd, codec_dai,
+						      asoc_rtd_to_cpu(rtd, cpu_id));
```

Comment:
@bardliao is it OK if the condition ```if (!rtd->dai_link->codec_ch_maps)``` is true in this block?

Reply:
> @bardliao is it OK if the condition `if (!rtd->dai_link->codec_ch_maps)` is true in this block?

No, and that's why it is in the 

else if (rtd->dai_link->num_codecs > rtd->dai_link->num_cpus &&
			   **rtd->dai_link->codec_ch_maps**)

block.

Reply:
ok then why is it relevant in the else block because n > M is not supported anyway right?

Reply:
> ok then why is it relevant in the else block because n > M is not supported anyway right?

```
                        dev_err(card->dev,
                                "N cpus to M codecs link is not supported: ");
                        if (!rtd->dai_link->codec_ch_maps)
                                dev_err(card->dev, "no mapping func provided\n");
                        else
                                dev_err(card->dev, "codec number %d < cpu number %d\n",
                                        rtd->dai_link->num_codecs, rtd->dai_link->num_cpus);
```
The `if (!rtd->dai_link->codec_ch_maps) else` here is just to tell the reason why N cpus to M codecs link is not supported.

Reply:
@bardliao I think you're confusing my ask. What I''m trying to say is that if N < M, then the check ```if (!rtd->dai_link->codec_ch_maps)``` is relevant. If not, it isnt. So if you club in the else block you need to check that condition along with N < M to call out that no mapping function has been provided. Otherwise, simply say that N > M is not supported whether or not there's mapping function provided.

Reply:
Thanks @ranj063 I got your point now. And I updated my code, thanks.